redis/tests/test_cluster_multi_node_reconnect.rs

```diff
@@ -0,0 +1,152 @@
+#![cfg(all(feature = "cluster-async", feature = "tokio-comp"))]
+
+mod support;
+
+use redis::{
+    cluster::ClusterClient,
+    cluster_routing::{RoutingInfo, SingleNodeRoutingInfo},
+    cmd, Value,
+};
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+use support::*;
+
+#[test]
+fn test_cluster_multi_node_disconnect_reconnects_successfully() {
```

Comment:
this test passes without your changes, too, so it doesn't actually check this change.